consensus/poet/core/tests/test_consensus/test_poet_block_publisher.py

```diff
@@ -674,3 +675,233 @@ def test_block_publisher_success_case(
+                'ConsensusStateStore')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.factory')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.'
+                'ConsensusState')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.'
+                'ValidatorRegistryView')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.utils')
+    def test_block_publisher_claim_readiness(
+            self,
+            mock_utils,
+            mock_validator_registry_view,
+            mock_consensus_state,
+            mock_poet_enclave_factory,
+            mock_consensus_state_store,
+            mock_poet_key_state_store,
+            mock_signup_info,
+            mock_poet_config_view,
+            mock_block_wrapper,
+            mock_wait_time):
+
+        """ Test verifies that PoET Block Publisher only claims
+         readiness if the wait timer has expired
+        """
+
+        # create a mock_validator_registry_view with
+        # get_validator_info that does nothing
+        mock_validator_registry_view.return_value.get_validator_info. \
+            return_value = \
+            ValidatorInfo(
+                name='validator_001',
+                id='validator_deadbeef',
+                signup_info=SignUpInfo(
+                    poet_public_key='00112233445566778899aabbccddeeff'))
+
+        # create a mock_consensus_state that returns a mock with
+        # the following settings:
+        mock_state = mock.Mock()
+        mock_state.validator_signup_was_committed_too_late.return_value = False
+        mock_state.validator_has_claimed_block_limit.return_value = False
+        mock_state.validator_is_claiming_too_early.return_value = False
+        mock_state.validator_is_claiming_too_frequently.return_value = False
+
+        mock_consensus_state.consensus_state_for_block_id.return_value = \
+            mock_state
+
+        mock_consensus_state_store.return_value.__getitem__.return_value = \
+            mock_consensus_state
+
+        # create mock_signup_info
+        mock_signup_info.unseal_signup_data.return_value = \
+            '00112233445566778899aabbccddeeff'
+
+        # create mock_batch_publisher
+        mock_batch_publisher = mock.Mock(
+            identity_signing_key=signing.generate_privkey())
+
+        mock_block_cache = mock.MagicMock()
+        mock_state_view_factory = mock.Mock()
+
+        # create mock_block_header with the following fields
+        mock_block = mock.Mock(identifier='0123456789abcdefedcba9876543210')
+        mock_block.header.signer_pubkey = '90834587139405781349807435098745'
+        mock_block.header.previous_block_id = '2'
+        mock_block.header.block_num = 1
+        mock_block.header.state_root_hash = '6'
+        mock_block.header.batch_ids = '4'
+
+        # create a mock_wait_timer that has expired
+        mock_wait_time = mock.Mock()
```

Comment:
You are re-assigning to the `mock_wait_time` that references the mock `WaitTimer` class.  This doesn't do what you expect.  Maybe rename the previous one to `mock_wait_timer_class` or rename your mock wait timer.